crates/forge_analyzer/src/interp.rs

```diff
@@ -676,9 +564,190 @@ impl<'cx, C: Runner<'cx>> Interp<'cx, C> {
+        match rvalue {
+            Rvalue::Read(operand) => self.value_from_operand(defid_block, operand),
+            Rvalue::Template(template) => {
+                let all_values = template
+                    .exprs
+                    .iter()
+                    .map(|expr| self.value_from_operand(defid_block, expr.clone()))
+                    .collect_vec();
+                if all_values.contains(&Value::Unknown) {
+                    return Value::Unknown;
+                }
+
+                let quasis_as_values: Vec<Value> = template
+                    .quasis
+                    .iter()
+                    .map(|quasis| Value::Const(Const::Literal(quasis.to_string())))
+                    .collect_vec();
+
+                let values_joined = quasis_as_values
+                    .iter()
+                    .zip(all_values.iter())
+                    .flat_map(|(a, b)| vec![a.clone(), b.clone()])
+                    .chain(quasis_as_values.iter().skip(all_values.len()).cloned())
+                    .chain(all_values.iter().skip(quasis_as_values.len()).cloned())
+                    .collect_vec();
+
+                return_combinations_phi(values_joined)
+            }
+            Rvalue::Bin(BinOp::Add, op1, op2) => {
+                let value_op1 = self.value_from_operand(defid_block, op1);
+                let value_op2 = self.value_from_operand(defid_block, op2);
+                if value_op1 == Value::Unknown || value_op2 == Value::Unknown {
+                    return Value::Unknown;
+                }
+                return_combinations_phi(vec![value_op1, value_op2])
+            }
+            _ => Value::Unknown,
+        }
+    }
+
+    #[inline]
+    fn value_from_operand(&self, defid_block: DefId, operand: Operand) -> Value {
+        match operand {
+            Operand::Var(Variable {
+                base: Base::Var(varid),
+                projections,
+            }) => {
+                let (varid, projections) = self.get_farthest_obj(defid_block, varid, projections);
+                if self.is_obj(varid) {
+                    return Value::Object(varid);
+                }
+                match self.get_value(defid_block, varid, Some(projections)) {
+                    Some(value) => value.clone(),
+                    None => Value::Unknown,
+                }
+            }
+            Operand::Lit(str) => {
+                if let Some(value) = convert_lit_to_raw(&str) {
+                    Value::Const(Const::Literal(value))
+                } else {
+                    Value::Unknown
+                }
+            }
+            _ => Value::Unknown,
+        }
+    }
+
+    #[inline]
+    fn get_farthest_obj(
+        &self,
+        defid_block: DefId,
+        varid: VarId,
+        mut projections: ProjectionVec,
+    ) -> (VarId, ProjectionVec) {
+        let mut current_var_id = varid;
+        for (i, _) in projections.clone().iter().enumerate() {
```

Comment:
I needed to clone the projections here, not sure if there is a way around it, since we return the projections element.

Reply:
Not for the loop, since you're only using the index. Instead of doing `for (i, _) in projections.iter().enumerate()`, you can just loop over the indices directly, i.e. `0..projections.len()`.

Reply:
fixed. Thank you!